keyboards/helix/rev2/keymaps/fraanrosi/keymap.c

```diff
@@ -0,0 +1,372 @@
+                } else {
+                    TOG_STATUS = !TOG_STATUS;
+                    #ifdef RGBLIGHT_ENABLE
+                        if (!caps_is_active) {
+                            RGB_current_mode = rgblight_config.mode;
+                            CURRENT_COLOR    = (HSV){rgblight_config.hue, rgblight_config.sat, rgblight_config.val};
+                        }
+                        rgblight_mode(RGBLIGHT_MODE_STATIC_LIGHT);
+                        rgblight_sethsv(HSV_BLUE);
+                    #endif
+                }
+                layer_on(_LOWER);
+            } else {
+                #ifdef RGBLIGHT_ENABLE
+                    if (!caps_is_active) {
+                        rgblight_mode(RGB_current_mode);
+                        rgblight_sethsv(CURRENT_COLOR.h, CURRENT_COLOR.s, CURRENT_COLOR.v);
+                    } else {
+                        rgblight_mode(RGBLIGHT_MODE_ALTERNATING);
+                        rgblight_sethsv(HSV_WHITE);
+                    }
+                #endif
+                TOG_STATUS = false;
+                layer_off(_LOWER);
+            }
+            return false;
+            break;
+        case RAISE:
+            if (record->event.pressed) {
+                // not sure how to have keyboard check mode and set it to a variable, so my work around
+                // uses another variable that would be set to true after the first time a reactive key is pressed.
+                if (TOG_STATUS) {  // TOG_STATUS checks is another reactive key currently pressed, only changes RGB mode if returns false
+                } else {
+                    TOG_STATUS = !TOG_STATUS;
+                    #ifdef RGBLIGHT_ENABLE
+                        if (!caps_is_active) {
+                            RGB_current_mode = rgblight_config.mode;
+                            CURRENT_COLOR    = (HSV){rgblight_config.hue, rgblight_config.sat, rgblight_config.val};
+                        }
+                        rgblight_mode(RGBLIGHT_MODE_STATIC_LIGHT);
+                        rgblight_sethsv(HSV_RED);
+                    #endif
+                }
+                layer_on(_RAISE);
+            } else {
+                #ifdef RGBLIGHT_ENABLE
+                    if (!caps_is_active) {
+                        rgblight_mode(RGB_current_mode);
+                        rgblight_sethsv(CURRENT_COLOR.h, CURRENT_COLOR.s, CURRENT_COLOR.v);
+                    } else {
+                        rgblight_mode(RGBLIGHT_MODE_ALTERNATING);
+                        rgblight_sethsv(HSV_WHITE);
+                    }
+                #endif
+                layer_off(_RAISE);
+                TOG_STATUS = false;
+            }
+            return false;
+            break;
+            // led operations - RGB mode change now updates the RGB_current_mode to allow the right RGB mode to be set after reactive keys are released
+        case RGB_MOD:
```

Comment:
For all the rgb modes, you could just do
```
        case RGB_MOD:
        case RGB_RMOD:
        case RGB_MODE_PLAIN ... RGB_MODE_TEST:
            if (!record->event.pressed) {
                RGB_current_mode = rgblight_get_mode();
            }
            break;

```

Reply:
Hi, 
The changes were made.
At this point, I couldn't just use that path because I put custom presets, but I tried a workaround.
Let me know if it's ok.
Regards!